gui/notify.lua

```diff
@@ -21,6 +21,22 @@ NotifyOverlay.ATTRS{
 function NotifyOverlay:init()
     self.state = {}
 
+    local on_submit
+    local on_submit2
+    
+    if not dfhack.world.isAdventureMode() then
+        on_submit = function(_, choice)
+            if not choice.data.on_click then return end
+            local prev_state = self.state[choice.data.name]
+            self.state[choice.data.name] = choice.data.on_click(prev_state, false)
+        end
+        on_submit2 = function(_, choice)
+            if not choice.data.on_click then return end
+            local prev_state = self.state[choice.data.name]
+            self.state[choice.data.name] = choice.data.on_click(prev_state, true)
+        end
+    end
```

Comment:
there are two problems here:
1. it will not have the intended effect for adventure mode since `init` runs at program start (when `dfhack.world.isAdventureMode()` is false
2. it removes click to zoom functionality in addition to enter handling

instead, you can leave `init` unchanged and add a `NotifyOverlay:onInput` function that intercepts the enter (`keys.SELECT`) key and returns false before calling `return NotifyOverlay.super.onInput(self, keys)`

I'd be ok with ignoring SELECT regardless of mode so the fort mode panel behaves the same as the adventure mode panel.